Cython/Compiler/Pipeline.py

```diff
@@ -21,6 +21,52 @@ def abort_on_errors(node):
         raise AbortError("pipeline break")
     return node
 
+def create_shared_library_pipeline(context, scope, options, result):
+
+    def generate_tree_factory(context):
+        def generate_tree(compsrc):
+            from .Code import TempitaUtilityCode
+            from .UtilityCode import CythonUtilityCode
+            from . import MemoryView
+            import os.path
+            import Cython
+
+            # Force to generate __Pyx_ExportFunction
+            Options.cimport_from_pyx = True
+
+            source_desc = compsrc.source_desc
+            full_module_name = compsrc.full_module_name
+
+
+            initial_pos = (source_desc, 1, 0)
+            scope = context.find_module(full_module_name, pos = initial_pos, need_pxd = 0)
+
+            tree = context.get_empty_tree(source_desc, scope, pxd=False, full_module_name=full_module_name)
+
+            tree.is_pxd = False
+            tree.compilation_source = compsrc
+            tree.scope = scope
+
+            scope.use_utility_code(MemoryView.shared_view_utility_code)
```

Comment:
I haven't completely thought this through (so it may be nonsense) but:

some of the issues with `internal` come from loading this as utility code and doing `merge_in`.

I wonder if it'd be better to treat this code as the source of the module rather than as utility code?

Reply:
I wasn't really suggesting multiple .so files - I think we should target just one. I'm was just wondering if there's a way to compile the code without treating it as "utility code" and so making everything internal.

Reply:
Haven't looked into the details either, but generally speaking, we should keep this as close to a regular extension module (and its build procedure) as possible.